core/spec/models/spree/order_shipping_method_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'spec_helper'
+
+describe Spree::Order do
+  let(:order) { create(:order_with_line_items) }
+  let(:quadcopter) { create(:shipping_method, name: "Quads'r'us", code: "QUADSRUS", cost: 200) }
+
+  it "preserves selected shipping method when update_cart and next! called" do
```

Comment:
I'd appreciate a 2nd set of eyes (and corresponding brain) to help ensure the test cases here are accurate.